docs: remove extraneous parts of DHIS2 Docker tutorial (#105)
dhis2/developer-portal: docs/tutorials/dhis2-docker.md

---
id: dhis2-docker
title: DHIS2 in Docker
---

## Prerequisites

Make sure you have the following tools installed
* [Docker](https://docs.docker.com/get-docker/)
* [Docker Compose](https://docs.docker.com/compose/install/)
* [d2 cluster CLI](https://cli.dhis2.nu/#/commands/d2-cluster) (_only needed if you want to use `d2 cluster` for running DHIS2 in Docker_)

## Overview

- [Run DHIS2 in Docker with Docker Compose](#run-dhis2-in-docker-with-docker-compose)
- [Run DHIS2 in Docker with d2 cluster CLI](#run-dhis2-in-docker-with-d2-cluster-cli)

## Run DHIS2 in Docker with Docker Compose

Running DHIS2 with Docker Compose is very straightforward, [see the dhis2-core README](https://github.com/dhis2/dhis2-core#run-dhis2-in-docker) for more details. 
```shell
DHIS2_IMAGE=dhis2/core:2.39.0.1 DHIS2_DB_DUMP=https://databases.dhis2.org/sierra-leone/2.39.0/dhis2-db-sierra-leone.sql.gz docker compose up
```

> This is the preferred way if you already have the `dhis2-core` repository cloned and you want a simple setup that consist of DHIS2 and a database.

Reviewer: ```suggestion
This is the preferred way if you want a simple setup that consist of DHIS2 and a database. You can either clone the `dhis2-core` repository or download only the `[docker-compose](https://github.com/dhis2/dhis2-core/blob/master/docker-compose.yml)` file.
```

## Run DHIS2 in Docker with d2 cluster CLI

Running DHIS2 with the `d2 cluster` CLI is also very simple, [see the DHIS2 CLI docs](https://cli.dhis2.nu/#/commands/d2-cluster) for more details.

```shell
d2 cluster up 2.39.0.1 --db-version 2.39 --seed
```

This way of running DHIS2 provides extra features like running multiple DHIS2 "clusters" at once, saving different configuration combinations and more.

---

> [//]: # (TODO Maybe the following should be moved to the D2 CLI docs)

Reviewer: I think it should be removed as its in https://cli.dhis2.nu/#/getting-started
and in https://developers.dhis2.org/docs/tutorials/setup-env 😅https://cli.dhis2.nu/#/getting-started

- [DHIS2 in Docker in production](#dhis2-in-docker-in-production)

## Run DHIS2 in Docker with Docker Compose

Running DHIS2 with Docker Compose is very straightforward, [see the dhis2-core README](https://github.com/dhis2/dhis2-core#run-dhis2-in-docker) for more details. 
```shell
DHIS2_IMAGE=dhis2/core:2.39.0.1 DHIS2_DB_DUMP=https://databases.dhis2.org/sierra-leone/2.39.0/dhis2-db-sierra-leone.sql.gz docker compose up
```

This is the preferred way if you want a simple setup that consist of DHIS2 and a database. You can either clone the [`dhis2-core`](https://github.com/dhis2/dhis2-core) repository or download only the [`docker-compose.yml`](https://github.com/dhis2/dhis2-core/blob/master/docker-compose.yml) file.

## Run DHIS2 in Docker with d2 cluster CLI

Running DHIS2 with the `d2 cluster` CLI is also very simple, [see the DHIS2 CLI docs](https://cli.dhis2.nu/#/commands/d2-cluster) for more details.

```shell
d2 cluster up 2.39.0.1 --db-version 2.39 --seed
```

This way of running DHIS2 provides extra features like running multiple DHIS2 "clusters" at once, saving different configuration combinations and more.

> ## DHIS2 in Docker in production

Reviewer: nit: newline after titles

> We can't recommend using [the DHIS2 Docker images](https://github.com/dhis2/dhis2-core#pre-built-images) in production yet. At this point we don’t have enough experience and therefore can’t vouch for their stability for “mission critical” production use. We are not saying that someone experienced with running Docker in production shouldn’t use them.  

Reviewer: This looks like the "old" sentence. Should we use the updated one that we iterated on in slack the other day?

title: DHIS2 Docker
---

## Overview

-   Install D2 CLI
-   Launch development cluster
-   System cleanup (_Optional_)

## Install D2 CLI

> 

Reviewer: I think we should always just refer to the "most authoritative" source in order for us to have docs that are up-to-date.

I suggest we add a "Prerequisites" section. Maybe

```markdown
## Prerequisites

Make sure you have the following tools installed

* [Docker](https://www.docker.com/)
* [Docker Compose](https://docs.docker.com/compose/)
* [DHIS2 d2 CLI](https://cli.dhis2.nu/#/getting-started) - only needed if you want to use `d2` for setting up DHIS2 in Docker.
```

We can also move the link to `d2` at the top of a `d2` section. In case we want to split the docs into `Docker Compose` and `d2`.
